presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java

```diff
@@ -298,13 +280,10 @@ public boolean apply(Node node)
                 // consider a split with a host hint without a port as being accessible
                 // by all nodes in that host
                 if (!hint.hasPort() || split.isRemotelyAccessible()) {
-                    for (Node node : nodeMap.getNodesByHost().get(address)) {
-                        if (includeCoordinator || !coordinatorIdentifier.equals(node.getNodeIdentifier())) {
-                            if (chosen.add(node)) {
-                                scheduleLocal.incrementAndGet();
-                            }
-                        }
-                    }
+                    nodeMap.getNodesByHost().get(address).stream()
+                            .filter(node -> includeCoordinator || !coordinatorIdentifier.equals(node.getNodeIdentifier()))
```

Comment:
This has the same logic as the one above. Too bad you can't save and apply the same stream logic to both.